ghmirror/data_structures/monostate.py

```diff
@@ -60,6 +63,18 @@ def __setitem__(self, key, value):
     def __iter__(self):
         return iter(self._data)
 
+    def __len__(self):
+        return len(self._data)
+
+    def __sizeof__(self):
+        """ Calculate the size of the dictionary and all its contents
```

Comment:
I'm thinking... We don't have to save the data pickled. And the unpickling overhead is an unwanted side effect.
What about changing the self._data dictionary so is looks like this:
```
{key: { 'size': 123, 'data': ResponseObject}}
```
The size would be calculated considering the pickled object and stored on __setitem__ . The __getitem__ would then return self._dict[key]['size'].

The __sizeof__ would use that info to calculate the overall size.

Wdyt?

Reply:
The getitem would still return self._dict[key]['data'] but sizeof would use self._dict[key]['size']. I like this approach because it beats just serializing at the metrics endpoint or storing serialized data. But will the size field include the size of both the key and value? Is that fine as long as it's clearly documented as such?

Reply:
Yep, sounds like a plan :)

Reply:
Correction... getitem would then return self._dict[key]['data']